src/android/LocalNotification.java

```diff
@@ -132,57 +158,228 @@ public boolean execute (final String action, final JSONArray args,
                     request(command);
-                } else
-                if (action.equals("actions")) {
+                } else if (action.equals("actions")) {
                     actions(args, command);
-                } else
-                if (action.equals("schedule")) {
+                } else if (action.equals("schedule")) {
                     schedule(args, command);
-                } else
-                if (action.equals("update")) {
+                } else if (action.equals("update")) {
                     update(args, command);
-                } else
-                if (action.equals("cancel")) {
+                } else if (action.equals("cancel")) {
                     cancel(args, command);
-                } else
-                if (action.equals("cancelAll")) {
+                } else if (action.equals("cancelAll")) {
                     cancelAll(command);
-                } else
-                if (action.equals("clear")) {
+                } else if (action.equals("clear")) {
                     clear(args, command);
-                } else
-                if (action.equals("clearAll")) {
+                } else if (action.equals("clearAll")) {
                     clearAll(command);
-                } else
-                if (action.equals("type")) {
+                } else if (action.equals("type")) {
                     type(args, command);
-                } else
-                if (action.equals("ids")) {
+                } else if (action.equals("ids")) {
                     ids(args, command);
-                } else
-                if (action.equals("notification")) {
+                } else if (action.equals("notification")) {
                     notification(args, command);
-                } else
-                if (action.equals("notifications")) {
+                } else if (action.equals("notifications")) {
                     notifications(args, command);
+                } else if (action.equals("hasDoNotDisturbPermissions")) {
+                    hasDoNotDisturbPermissions(command);
+                } else if (action.equals("requestDoNotDisturbPermissions")) {
+                    requestDoNotDisturbPermissions(command);
+                } else if (action.equals("isIgnoringBatteryOptimizations")) {
+                    isIgnoringBatteryOptimizations(command);
+                } else if (action.equals("requestIgnoreBatteryOptimizations")) {
+                    requestIgnoreBatteryOptimizations(command);
+                } else if (action.equals("dummyNotifications")) {
+                    dummyNotifications(command);
                 }
             }
         });
 
         return true;
     }
 
+    /**
+     * Required for Android 13 to get the runtime notification permissions.
+     *
+     * @param command The callback context used when calling back into JavaScript.
+     */
+    private void dummyNotifications(CallbackContext command) {
```

Comment:
This method is just creating a channel and should be renamed to reflect this.

Reply:
I agree it is not optimal, though I'd prefer to merge this first before revisiting the purpose of this function